src/processes/news-process.js

```diff
@@ -1,213 +1,250 @@
-            // await postToGetstream(activity, job.user_id);
-            message = 'url news not unique'
-        } else {
-
-            data.news_link_id = uuidv4();
-            data.url = data.news_url;
-
-            const site_name = data.site_name
-            const activity = {
-                domain: {
-                    name, site_name, info, image: logo, domain_page_id: data.domain_page_id
-                },
-                content: { ...data, ...dateCreted }
-            }
-
-            let result = await postToGetstream(activity, job.user_id);
-            await putMainFeed(job, name, logo, created_domain, data, result?.id);
-
-            let postId = result.returnActivityId
-            console.info(result);
-            console.info(`postId in Process : ${postId}`);
-
-            let newsLink = await NewsLink.create({ ...data, ...dateCreted, post_id: result.id })
-            await new ElasticNewsLink().putToIndexFromGetstreamObject(newsLink)
-            message = 'news link created'
+  try {
+    const findNewsLink = await NewsLink.findOne({
+      where: {news_url: data.news_url}
+    });
+    let message;
+    if (findNewsLink) {
+      console.info('id', findNewsLink.news_link_id);
+      data.news_link_id = findNewsLink.news_link_id;
+      data.url = data.news_url;
+
+      const {site_name} = data;
+      const activity = {
+        domain: {
+          name,
+          site_name,
+          info,
+          image: logo,
+          domain_page_id: data.domain_page_id
+        },
+        content: {
+          ...data,
+          ...dateCreted,
+          content_published_at: findNewsLink?.created_article || data?.content_published_at
         }
-
-        return message
-    } catch (error) {
-        return error
+      };
+
+      await putMainFeed(job, name, logo, created_domain, data, findNewsLink?.post_id);
+      await new ElasticNewsLink().putToIndexFromGetstreamObject(findNewsLink);
+      // await postToGetstream(activity, job.user_id);
+      message = 'url news not unique';
+    } else {
+      data.news_link_id = uuidv4();
+      data.url = data.news_url;
+
+      const {site_name} = data;
+      const activity = {
+        domain: {
+          name,
+          site_name,
+          info,
+          image: logo,
+          domain_page_id: data.domain_page_id
+        },
+        content: {
+          ...data,
+          ...dateCreted,
+          content_published_at: findNewsLink?.created_article || data?.content_published_at
+        }
+      };
+
+      const result = await postToGetstream(activity, job.user_id);
+      await putMainFeed(job, name, logo, created_domain, data, result?.id);
+
+      const postId = result.returnActivityId;
+      console.info(result);
+      console.info(`postId in Process : ${postId}`);
+
+      const newsLink = await NewsLink.create({
+        ...data,
+        ...dateCreted,
+        post_id: result.id,
+        created_article: data.contentPublishedAt
+      });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The property `contentPublishedAt` is being assigned to `created_article`. However, in the rest of the code, the camelCase version `content_published_at` is used. For consistency, it's better to stick with one naming convention.
```diff
-         created_article: data.contentPublishedAt
+         content_published_at: data.contentPublishedAt
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->

Reply:
will change

Reply:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

@fajarism Great! Consistency in naming conventions is crucial for code readability and maintainability. If you need any further assistance, feel free to ask.

<!-- This is an auto-generated reply by OSS CodeRabbit -->